torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -176,15 +176,15 @@ Variable & VariableType::checked_cast_variable(Tensor & t, const char * name, in
 }
 
 const Tensor & VariableType::unpack(const Tensor & t, const char * name, int pos) {
-  return checked_cast_variable(t, name, pos).data();
+  return checked_cast_variable(t, name, pos);
```

Comment:
It's okay to remove `.data()` here (and in the following `unpack()` variants), because the original purpose of `unpack()` is to obtain the Variable `t`'s underlying non-requires-grad data tensor, which is now unnecessary because the Variable `t` itself is the data tensor now.

We might ask "but this can return a requires-grad tensor instead of a non-requires-grad one, is that okay?" The answer is yes, because we always enable `at::AutoNonVariableTypeMode` before passing the "unpacked" tensors to the `baseType` dispatch functions, and with the thread-local mode enabled, the tensors will always be treated as non-requires-grad.

One notable use case that this change will break is the `VariableType::_sparse_coo_tensor_with_dims_and_tensors` function, which needs its own manual implementation after this patch, with the reasoning below.